### PR TITLE
Timeout agent connection

### PR DESCRIPTION
**Pull request by kvish**
Summary:
If the agent future failed to complete with exception we submit a kill request and throw the exception

JIRA issues: DCOS_OSS-5066

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java [hunk 1]
```diff
@@ -161,7 +161,12 @@ public boolean canProvision(Label label) {
                         node -> {
                           logger.info("Agent {} is online", name);
                           return node;
-                        });
+                        })
+                    .exceptionally(e -> {
+                      logger.info("Agent {} failed to come online", name);
```
Comment (kvish) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java]: @jeschkies this block is needed to provide a callback to kill the agent when the future completes with any exception

File: src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java [hunk 2]
```diff
@@ -287,6 +291,11 @@ private void updateState(StateEventOrSnapshot event) {
     }
   }
 
+  /** test method to set the agent timeout duration */
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java:294]: What do you mean by "test"?

**Review by jeschkies — APPROVED** [review]
Nice!

File: src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java (hunk 2 above)
Comment (kvish) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java:294]: ah just mean to be used for test purposes

**Review by takirala — APPROVED** [review]

**Pull request merged**